src/main/java/edu/kis/powp/jobs2d/drivers/DriverVisitor.java

```diff
@@ -0,0 +1,15 @@
+package edu.kis.powp.jobs2d.drivers;
+
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.jobs2d.drivers.monitoring.DriverMonitorDecorator;
+import edu.kis.powp.jobs2d.transformations.RotateTransformationDecorator;
+import edu.kis.powp.jobs2d.transformations.ScaleTransformationDecorator;
+
+public interface DriverVisitor {
+    void visit(ComplexDriver dirver);
+    void visit(InformativeLoggerDriver driver);
+    void visit(LineDriverAdapter driver);
+    void visit(RotateTransformationDecorator decorator);
+    void visit(ScaleTransformationDecorator decorator);
+    void visit(DriverMonitorDecorator decorator);
```

Comment:
I think that it would be best to introduce a common template for decorators (in the future), and redesign the visitor.